src/main/java/io/confluent/connect/elasticsearch/DataConverter.java

```diff
@@ -85,8 +86,8 @@ public DataConverter(ElasticsearchSinkConnectorConfig config) {
   }
 
   private String convertKey(Schema keySchema, Object key) {
-    if (key == null) {
-      throw new DataException("Key is used as document id and can not be null.");
+    if (key == null || String.valueOf(key).isEmpty()) {
```

Comment:
nit: Lets have two different checks and respective log lines

Reply:
updated